packages/js/src/plugins/candyMachineModule/operations/findCandyMachinesByPublicKeyField.ts

```diff
@@ -112,17 +112,31 @@ export const findCandyMachinesByPublicKeyFieldOperationHandler: OperationHandler
         .getMultipleAccounts(collectionPdas, commitment);
       scope.throwIfCanceled();
 
-      return zipMap(
+      const candyMachines = zipMap(
         unparsedAccounts,
         unparsedCollectionAccounts,
-        (unparsedAccount, unparsedCollectionAccount) => {
+        async (unparsedAccount, unparsedCollectionAccount) => {
           const account = parseCandyMachineAccount(unparsedAccount);
           const collectionAccount = unparsedCollectionAccount
             ? parseCandyMachineCollectionAccount(unparsedCollectionAccount)
             : null;
 
-          return toCandyMachine(account, unparsedAccount, collectionAccount);
+          const mint = account.data.tokenMint
+            ? await metaplex
+                .tokens()
+                .findMintByAddress({ address: account.data.tokenMint })
+                .run()
+            : null;
```

Comment:
Sadly, this is not performant as it will create an additional blockchain request for every SPL Token Candy Machine in the list. We should fetch all mint accounts at once using a `getMultipleAccounts` call and then reconciliate each mint account with its candy machine account.

Reply:
@lorisleiva thanks for the feedback Loris. I have a better understanding of what you were doing in that part of the codebase now.

I took another shot at the implementation and added an add'l test for parsing out the token mint. Let me know if this is more in line w/what you wanted